symfony/src/Service/SiteSyncService.php

```diff
@@ -0,0 +1,257 @@
+        if (!$preview) {
+            if ($sitesCount === 0) {
+                throw new \Exception('No sites found');
+            }
+            foreach ($deleted as $siteId) {
+                $site = $existingSites[$siteId];
+                $site->setDeleted(1);
+                $this->em->persist($siteData);
+                $this->em->flush();
+                $this->loggerService->log(Log::SITE_DELETE, $existingSites[$siteId]->getId());
+            }
+        }
+
+        return [
+            'created' => $created,
+            'modified' => $modified,
+            'deleted' => array_values($deleted)
+        ];
+    }
+
+    public function syncAwardees()
+    {
+        $entries = $this->getAwardeeEntriesFromRdr();
+        $awardeesMap = [];
+        foreach ($entries as $entry) {
+            $awardee = $entry->resource;
+            if (empty($awardee->id) || empty($awardee->displayName)) {
+                continue;
+            }
+            if ($awardee->id === 'UNSET') {
+                continue;
+            }
+            $awardeesMap[trim($awardee->id)] = $awardee->displayName;
+        }
+        if (empty($awardeesMap)) {
+            throw new \Exception('No awardees found');
+        }
+        $cmd = $this->em->getClassMetadata(Awardee::class);
+        $connection = $this->em->getConnection();
+        $connection->beginTransaction();
+        try {
+            $connection->query('DELETE FROM ' . $cmd->getTableName());
+            foreach ($awardeesMap as $id => $name) {
+                $awardee = new Awardee;
+                $awardee->setId($id);
+                $awardee->setName($name);
+                $this->em->persist($awardee);
+                $this->loggerService->log(Log::AWARDEE_ADD, $id);
+            }
+            $this->em->flush();
+            $connection->commit();
+        } catch (\Exception $e) {
+            $connection->rollback();
+            throw $e;
+        }
+    }
+
+    public function syncOrganizations()
+    {
+        $entries = $this->getAwardeeEntriesFromRdr();
+        $organizationsMap = [];
+        foreach ($entries as $entry) {
+            $awardee = $entry->resource;
+            if (!isset($awardee->organizations) || !is_array($awardee->organizations)) {
+                continue;
+            }
+            foreach ($awardee->organizations as $organization) {
+                if (empty($organization->id) || empty($organization->displayName)) {
+                    continue;
+                }
+                $organizationsMap[trim($organization->id)] = $organization->displayName;
+            }
+        }
+        if (empty($organizationsMap)) {
+            throw new \Exception('No organizations found');
+        }
+        $cmd = $this->em->getClassMetadata(Organization::class);
+        $connection = $this->em->getConnection();
+        $connection->beginTransaction();
+        try {
+            $connection->query('DELETE FROM ' . $cmd->getTableName());
```

Comment:
Can you use the query builder or DQL to perform this delete instead of using raw SQL?